### PR TITLE
CB-11364 Use immediate RxJava scheduler in tests in order to prevent …

### PR DESCRIPTION
**Pull request by lturcsanyi**
…flaky tests

See detailed description in the commit message.

### CODE REVIEW

**Review by lnardai — APPROVED** [review]

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import java.util.Map;
 
 import org.jetbrains.annotations.NotNull;
```
Comment (lajosrodek) [inline cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java:17]: I think the agreement in Slack was not to use these JetBrains annotations at all.

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
...
+import org.junit.After;
```
Comment (lajosrodek) [inline cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java:18]: Could you please migrate this class to JUnit 5 if possible?

**Review by lajosrodek — COMMENTED** [review]
My unwanted 2 cents. 😃

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java (hunk 1 above)
Comment (lturcsanyi) [reply cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java]: Sorry, my bad, I will fix it.

File: cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java (hunk 2 above)
Comment (lturcsanyi) [reply cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java]: Yes

Comment (lajosrodek) [reply cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtilsTest.java]: Awesome, thanks!

**Pull request merged**